visualization/app/codeCharta/util/indexedDB/indexedDBWriter.spec.ts

```diff
@@ -0,0 +1,72 @@
+import "fake-indexeddb/auto"
+import { openDB } from "idb"
+import { defaultState } from "../../state/store/state.manager"
+import {
+	CCSTATE_PRIMARY_KEY,
+	CCSTATE_STATE_ID,
+	CCSTATE_STORE_NAME,
+	DB_NAME,
+	DB_VERSION,
+	readCcState,
+	writeCcState
+} from "./indexedDBWriter"
+
+describe("IndexedDBWriter", () => {
+	beforeEach(() => {
+		jest.clearAllMocks()
+	})
+
+	describe("writeCcState", () => {
+		it("should successfully write state to the database", async () => {
+			await writeCcState(defaultState)
+
+			const database = await openDB(DB_NAME, DB_VERSION)
+			const transaction = database.transaction(CCSTATE_STORE_NAME, "readonly")
+			const store = transaction.objectStore(CCSTATE_STORE_NAME)
+			const result = await store.get(CCSTATE_STATE_ID)
+
+			expect(result.state).toEqual(defaultState)
+
+			database.close()
```

Comment:
here the database is closed after the expect, in the tests below it is closed before the expect, both should be handled equally